packages/state-transition/src/util/genesis.ts

```diff
@@ -127,6 +129,7 @@ export function applyTimestamp(config: ChainForkConfig, state: CachedBeaconState
  * @returns active validator indices
  */
 export function applyDeposits(
+  fork: ForkSeq,
```

Comment:
a lot of our code has new "fork" param and it caused a lot of files changed, we can just get the config from state and determine the fork based on state.slot
the good thing is we don't have to keep computing fork in different functions if we pass fork param
I think we can store fork inside the CachedBeaconState so that we don't have to pass fork param around or keep recomputing it in different functions